vault/identity_store_oidc_provider.go

```diff
@@ -1563,6 +1579,33 @@ func (i *IdentityStore) pathOIDCAuthorize(ctx context.Context, req *logical.Requ
 		scopes:      scopes,
 	}
 
+	// Validate the optional Proof Key for Code Exchange (PKCE) if a code
```

Comment:
We could also have an option for enforcing PKCE for a given client ID. Happy to consider adding it if others agree.

Reply:
That might be a useful feature for OIDC Provider. We recently got feedback from an auth/jwt user that PKCE was crucial for meeting "strict security requirements".

It is something we could always add at a later date as well.

Reply:
I agree. I think doing it now makes sense, since it shouldn't be a large change.

To @calvn's comment below - I'm wondering if this configuration should be at a client or server level. I _think_ client level makes the most sense, since the application environment can vary (e.g., confidential, public), and PKCE is focused on protecting public clients.

Reply:
Client-level should be fine since this is an opt-in and something that individual client owners should determine. We could add a toggle on the server level down the road if needed.

Reply:
Resolving this since we decided to not have a separate configuration to enforce PKCE for now. We can add it if there is a feature request.